src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable no-param-reassign */
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path');
+const { DefinePlugin } = require('webpack');
+const fs = require('fs');
+
+const DirectorySwitcherPlugin = require('./resolve/DirectorySwitcherPlugin');
+const LocalPackageAliasPlugin = require('./resolve/LocalPackageAliasPlugin');
+const { babelLoader, mdxOptions, getMdxLoader } = require('./siteLoaderUtils');
+const getNewRelicJS = require('../new-relic/getNewRelicJS');
+
+// Singletons
+let oneTimeSetupComplete = false;
+const siteRegistry = {};
+const processPath = process.cwd();
+const isLernaMonoRepo = fs.existsSync(path.join(processPath, 'lerna.json'));
+
+/**
+ * Updates the webpack options with defaults that terra-dev-site requires.
+ */
+class SitePlugin {
+  constructor({
+    entry,
+    config,
+    applyDefaults,
+    contentDirectory,
+  } = {}) {
+    // Apply defaults to the config.
+    this.siteConfig = applyDefaults(config);
```

Comment:
What happens if applyDefaults is undefined?

Reply:
It should blow up, applyDefaults is required.